server/controllers/userController.js

```diff
@@ -113,94 +123,143 @@ const UserController = {
         }
       ]
     };
+    query.attributes = userDetails.user;
+    query.limit = request.query.limit || null;
+    query.offset = request.query.offset || null;
+    query.order = [['createdAt', 'DESC']];
     db.users
-      .findAll(query)
+      .findAll({ query: query, limit: query.limit, offset: query.offset })
       .then((allUsers) => {
         if (allUsers) {
-          res.status(200).send({
+          response.status(200).send({
             message: 'Listing available users',
             data: allUsers
           });
         }
       })
       .catch((err) => {
-        res.status(404).send({
-          message: 'There was a problem getting all users'
+        response.status(400).send({
+          message: 'There was a problem getting all users',
+          err
         });
       });
   },
 
-  updateUser(req, res) {
+  updateUser(request, response) {
     db.users
-      .findById(req.params.id)
+      .findById(request.params.id)
       .then((user) => {
         if (user) {
-          if (toString(req.decodedToken.userId) !== toString(req.params.id)) {
-            return res.send({ message: 'Request not allowed' });
+          if (String(request.decodedToken.userId) !== String(request.params.id)) {
+            return response.send({ message: 'Request not allowed' });
           }
           user.update({
-            firstName: req.body.firstName || user.firstName,
-            lastName: req.body.lastName || user.lastName,
-            email: req.body.email || user.email,
-            username: req.body.username || user.username,
-            password: req.body.password || user.password
+            firstName: request.body.firstName || user.firstName,
+            lastName: request.body.lastName || user.lastName,
+            email: request.body.email || user.email,
+            userName: request.body.userName || user.userName,
+            password: request.body.password || user.password
```

Comment:
Your user model triggers a beforeUpdate that rehashes your password. What do you think will happen here if it comes across your already hashed password.
Will it rehash it or just resave it? what would be the new password if it rehashes the hash of your password